MyObjects/Helpers/MyWebDriver.cs

```diff
@@ -0,0 +1,192 @@
+
+            if (status == TestStatus.Passed)
+            {
+                Logger.Info($"Screenshot: {description}");
+            }
+            else
+            {
+                Logger.Error($"Screenshot: {description}");
+            }
+        }
+
+        /// <summary>
+        /// Hard Assert: actual Url against expected
+        /// </summary>
+        /// <param name="urlExpected"></param>
+        public void AssertUrl(string urlExpected)
+        {
+            string urlActual = webDriver.Url;
+            bool isMatch = urlActual.Equals(urlExpected);
+            if (isMatch)
+            {
+                Logger.Info($"PASS: Actual URL: '{urlActual}' is equal to the Expected URL: '{urlExpected}'");
+            }
+            else
+            {
+                Logger.Error($"FAIL: Actual URL: '{urlActual}' is not equal to the Expected URL: '{urlExpected}'");
+                Assert.Fail();
+            }
+
+        }
+
+        /// <summary>
+        /// Hard Assert: verify element is present on page
+        /// </summary>
+        /// <param name="locator"></param>
+        public void AssertElementPresentOnPage(By locator)
+        {
+            IWebElement element = FindElement(locator);
+            if (element != null)
+            {
+                Logger.Info($"PASS: Expected object with locator: {locator} is present on page");
+
+            }
+            else
+            {
+                Logger.Error($"FAIL: Expected object with locator: {locator} is absent on page");
+                Assert.Fail();
+            }
+        }
+
+        /// <summary>
+        /// Soft Assert: verify text is present on page
+        /// </summary>
+        /// <param name="text"></param>
+        public void AssertTextPresentOnPage(string text)
+        {
+            if (webDriver.PageSource.Contains(text))
+            {
+                Logger.Info($"PASS: Expected text: {text} is present on page");
+            }
+            else
+            {
+                Logger.Error($"FAIL: Expected text: {text} is absent on page");
+
+            }
+        }
+
+        /// <summary>
+        /// Soft Assert: verify number of items on page against expected
+        /// </summary>
+        /// <param name="locator"></param>
+        /// <param name="expectedCount"></param>
+        public void AssertCountOfItemsOnPage(By locator, int expectedCount)
+        {
+            ReadOnlyCollection<IWebElement> list = FindElements(locator);
+            int itemsCount = list.Count;
+            if(itemsCount == expectedCount)
+            {
+                Logger.Info($"PASS: Actual count of items on page is {expectedCount}");
+            }
+            else
+            {
+                Logger.Info($"FAIL: Actual count of items on page is not equal to {expectedCount}");
+            }
+        }
+    }
```

Comment:
1. pls add one more comparison assert that'd accept expected, actual and description parameters
2. pls also add soft asserts